sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveries.java

```diff
@@ -0,0 +1,730 @@
+ * which the application wishes to occur at the broker. The broker acknowledges this with a disposition frame
+ * with a state (a.k.a. remote-state) representing the actual outcome (a.k.a. remote-outcome) of any work
+ * the broker performed upon processing the settlement request and a flag (a.k.a. remotely-settled) indicating
+ * whether the broker settled the delivery.
+ */
+public final class ReceiverUnsettledDeliveries implements AutoCloseable {
+    // Ideally value of this const should be 'deliveryTag' but given the only use case today is as Service Bus
+    // LockToken, while logging, use the value 'lockToken' to ease log parsing.
+    // (TODO: anuchan; consider parametrizing the value of deliveryTag?).
+    private static final String DELIVERY_TAG_KEY = "lockToken";
+    private final AtomicBoolean isTerminated = new AtomicBoolean();
+    private final String hostName;
+    private final String entityPath;
+    private final String receiveLinkName;
+    private final ReactorDispatcher dispatcher;
+    private final AmqpRetryPolicy retryPolicy;
+    private final Duration timeout;
+    private final UUID deliveryEmptyTag;
+    private final ClientLogger logger;
+    private final Disposable timoutTimer;
+
+    // The deliveries received, for those the application haven't sent disposition frame to the broker requesting
+    // settlement or disposition frame is sent, but yet to receive acknowledgment disposition frame from
+    // the broker indicating the outcome (a.k.a. remote-outcome).
+    private final ConcurrentHashMap<String, Delivery> deliveries = new ConcurrentHashMap<>();
+    // A collection of work, where each work representing the disposition frame that the application sent,
+    // waiting to receive an acknowledgment disposition frame from the broker indicating the outcome
+    // (a.k.a. remote-outcome).
+    private final ConcurrentHashMap<String, DispositionWork> pendingDispositions = new ConcurrentHashMap<>();
+
+    /**
+     * Creates ReceiverUnsettledDeliveries.
+     *
+     * @param hostName         the name of the host hosting the messaging entity identified by {@code entityPath}.
+     * @param entityPath       the relative path identifying the messaging entity from which the deliveries are
+     *                         received from, the application can later disposition these deliveries by sending
+     *                         disposition frames to the broker.
+     * @param receiveLinkName  the name of the amqp receive-link 'Attach'-ed to the messaging entity from
+     *                         which the deliveries are received from.
+     * @param dispatcher       the dispatcher to invoke the ProtonJ library API to send disposition frame.
+     * @param retryOptions     the retry configuration to use when resending a disposition frame that the broker 'Rejected'.
+     * @param deliveryEmptyTag reference to static UUID indicating absence of delivery tag in deliveries.
```

Comment:
This parameter is nothing but a reference to MessageUtils.ZERO_LOCK_TOKEN, we'll remove this parameter once (in phase-2) we have ReceiveLinkHandler doing the decoding immediately upon delivery, i.e. ReceiveLinkHandler will be knowing how to parse the delivery tag.